core/src/main/java/org/apache/calcite/rel/externalize/RelJson.java

```diff
@@ -387,6 +394,12 @@ private Object toJson(RelDataType node) {
       if (node.getComponentType() != null) {
         map.put("component", toJson(node.getComponentType()));
       }
+      if (node.getKeyType() != null) {
+        map.put("key", toJson(requireNonNull(node.getKeyType(), "key type")));
+      }
+      if (node.getValueType() != null) {
+        map.put("value", toJson(requireNonNull(node.getValueType(), "value type")));
+      }
```

Comment:
`requireNonNull` would not be required if you save `getKeyType()` result to a variable

Reply:
Thank you for your kindly remind. It helps a lot.